src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/draftorders/upload/UploadDraftOrderAboutToSubmitHandlerTest.java

```diff
@@ -0,0 +1,300 @@
+        ).build();
+
+        caseData.getDraftOrdersWrapper().getUploadSuggestedDraftOrder().setUploadParty(uploadParty);
+
+        when(caseAssignedRoleService.getCaseAssignedUserRole(String.valueOf(caseID), AUTH_TOKEN))
+            .thenReturn(CaseAssignedUserRolesResource.builder().caseAssignedUserRoles(Collections.emptyList()).build());
+
+        UserInfo mockUserInfo = mock(UserInfo.class);
+        when(idamAuthService.getUserInfo(AUTH_TOKEN)).thenReturn(mockUserInfo);
+        when(mockUserInfo.getName()).thenReturn("Hamzah");
+
+        doNothing().when(draftOrdersCategoriser).categoriseDocuments(any(FinremCaseData.class), anyString());
+
+        when(draftOrderService.isOrdersSelected(List.of(ORDER_TYPE))).thenReturn(true);
+        when(draftOrderService.isPsaSelected(List.of(ORDER_TYPE))).thenReturn(false);
+
+        // When
+        GenericAboutToStartOrSubmitCallbackResponse<FinremCaseData> response =
+            handler.handle(FinremCallbackRequestFactory.from(1727874196328932L, caseData), AUTH_TOKEN);
+
+        // Then
+        List<SuggestedDraftOrderCollection> collectionResult = response.getData().getDraftOrdersWrapper().getSuggestedDraftOrderCollection();
+        Assertions.assertEquals(3, collectionResult.size());
+
+        SuggestedDraftOrder draftOrderResult1 = response.getData().getDraftOrdersWrapper().getSuggestedDraftOrderCollection().get(0).getValue();
+        assertThat(draftOrderResult1.getDraftOrder()).isNotNull();
+        Assertions.assertEquals(2, draftOrderResult1.getAttachments().size());
+
+        SuggestedDraftOrder draftOrderResult2 = response.getData().getDraftOrdersWrapper().getSuggestedDraftOrderCollection().get(1).getValue();
+        assertThat(draftOrderResult2.getDraftOrder()).isNotNull();
+        Assertions.assertEquals(1, draftOrderResult2.getAttachments().size());
+
+        SuggestedDraftOrder draftOrderResult3 = response.getData().getDraftOrdersWrapper().getSuggestedDraftOrderCollection().get(2).getValue();
+        assertThat(draftOrderResult3.getDraftOrder()).isNotNull();
+        assertThat(draftOrderResult3.getAttachments()).isNull();
+    }
+
+    @ParameterizedTest
+    @ValueSource(booleans = {true, false})
+    void shouldClearTemporaryFields(boolean scenario) {
+        DraftOrdersWrapper.DraftOrdersWrapperBuilder builder = DraftOrdersWrapper.builder();
+        if (scenario) {
+            builder.uploadAgreedDraftOrder(UploadAgreedDraftOrder.builder().build());
+        } else {
+            builder.uploadSuggestedDraftOrder(UploadSuggestedDraftOrder.builder().build());
+        }
+        FinremCaseData caseData = FinremCaseData.builder().draftOrdersWrapper(builder.build()).build();
+
+        GenericAboutToStartOrSubmitCallbackResponse<FinremCaseData> response =
+            handler.handle(FinremCallbackRequestFactory.from(1727874196328932L, caseData), AUTH_TOKEN);
+        assertThat(response.getData().getDraftOrdersWrapper().getUploadAgreedDraftOrder()).isNull();
+    }
```

Comment:
I'm_ pretty sure that two fields were cleared.  Apologies if I misremembered

Reply:
Additional field test added